lib/rbs/collection/config/lockfile_generator.rb

```diff
@@ -149,16 +162,37 @@ def generate
           end
         end
 
-        private def assign_stdlib(name:, from_gem:)
+        private def assign_stdlib(name:, from_gem: nil)
           return if lockfile.gems.key?(name)
 
           case name
           when 'rubygems', 'set'
             msg = "`#{name}` has been moved to core library, so it is always loaded. Remove explicit loading `#{name}`"
             msg << " from `#{from_gem}`" if from_gem
-            RBS.logger.warn msg
-
+            msg << "."
             return
+          when *ALUMNI_STDLIBS.keys
+            version = ALUMNI_STDLIBS.fetch(name)
+            if from_gem
+              # From `dependencies:` of a `manifest.yaml` of a gem
+              source = find_source(name: name) or raise
+              if source.is_a?(Sources::Stdlib)
+                RBS.logger.warn {
+                  "`#{name}` is included in the RBS dependencies of `#{from_gem}`, but the type definition as a stdlib in rbs-gem is deprecated. Add `#{name}` (#{version}) to the dependency of your Ruby program to use the gem-bundled type definition."
+                }
+              else
+                RBS.logger.info {
```

Comment:
This is `INFO` message because the end-user is not responsible for the `dependencies:` of the type definition.